app/src/main/java/com/alexandr4/loftcoin/screens/converter/ConverterFragment.java

```diff
@@ -0,0 +1,200 @@
+
+    @BindView(R.id.destination_amount)
+    TextView destinationAmount;
+
+    TextView sourceCurrencySymbolText;
+    TextView sourceCurrencySymbolName;
+
+    TextView destinationCurrencySymbolText;
+    TextView destinationCurrencySymbolName;
+
+    private ConverterViewModel viewModel;
+
+    private CompositeDisposable disposables = new CompositeDisposable();
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        viewModel.saveState(outState);
+        super.onSaveInstanceState(outState);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_converter, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        ButterKnife.bind(this, view);
+
+        Database database = ((App) getActivity().getApplication()).getDatabase();
+        viewModel = new ConverterViewModelImpl(savedInstanceState, database);
+
+        toolbar.setTitle(R.string.converter_screen_title);
+
+        sourceCurrencySymbolText = sourceCurrency.findViewById(R.id.symbol_text);
+        sourceCurrencySymbolName = sourceCurrency.findViewById(R.id.currency_name);
+
+        destinationCurrencySymbolText = destinationCurrency.findViewById(R.id.symbol_text);
+        destinationCurrencySymbolName = destinationCurrency.findViewById(R.id.currency_name);
+
+        if (savedInstanceState == null) {
+            sourceAmount.setText("1");
+        }
+
+        Fragment bottomSheetSource = getFragmentManager().findFragmentByTag(SOURCE_CURRENCY_BOTTOM_SHEET_TAG);
+        if (bottomSheetSource != null) {
+            ((CurrenciesBottomSheet) bottomSheetSource).setListener(sourceListener);
+        }
+
+        Fragment bottomSheetDestination = getFragmentManager().findFragmentByTag(DESTINATION_CURRENCY_BOTTOM_SHEET_TAG);
+        if (bottomSheetDestination != null) {
+            ((CurrenciesBottomSheet) bottomSheetDestination).setListener(destinationListner);
+        }
+        initOutputs();
+        initInputs();
+    }
+
+    @Override
+    public void onDestroyView() {
+        disposables.dispose();
+        super.onDestroyView();
+    }
+
+    private void initOutputs() {
+
+        Disposable disposable1 = RxTextView.afterTextChangeEvents(sourceAmount)
+                .subscribe(event -> viewModel.onSourceAmountChange(event.getEditable().toString()));
+
+        sourceCurrency.setOnClickListener(v -> viewModel.onSourceCurrencyClick());
+        destinationCurrency.setOnClickListener(v -> viewModel.onDestinationCurrencyClick());
+        disposables.add(disposable1);
+    }
+
+    private void initInputs() {
+
+        Disposable disposable1 = viewModel.sourceCurrency().subscribe(s ->
```

Comment:
could you name it more descriptive?

Reply:
### disposable1? no. here the character of currency and its text designation for convertible currency is established, but that returns to a variable disposable1 - for me not obviously